### DISCUSSION TITLE
follow-ups after 761

### DISCUSSION THREAD

**Discussion by jeromedockes** [skrub-data/skrub#793] [General]
a couple of discussions that started in @glemaitre 's  #761 but are not directly related and thus should not delay it:

- [ ] having the remainder columns in `transformer_` always be column names, not indices [comment](https://github.com/skrub-data/skrub/pull/761#discussion_r1343810885)
- [ ] or even not having a remainder anymore if we handle bools together with numerical columns [comment](https://github.com/skrub-data/skrub/pull/761#discussion_r1348594702)
- [ ] always exposing the `dtypes_` attribute, even when we don't autocast and thus `dtypes_` is identical to the input's `.dtypes` [comment](https://github.com/skrub-data/skrub/pull/761#discussion_r1346065690)

**Comment by Vincent-Maladiere** [comment]
> having the remainder columns in transformer_ always be column names, not indices https://github.com/skrub-data/skrub/pull/761#discussion_r1343810885

+1 and this is happening with https://github.com/scikit-learn/scikit-learn/issues/27533

> or even not having a remainder anymore if we handle bools together with numerical columns https://github.com/skrub-data/skrub/pull/761#discussion_r1348594702

+1 and I'm wondering if they are edge cases / rare types that we should also handle.


> always exposing the dtypes_ attribute, even when we don't autocast and thus dtypes_ is identical to the input's .dtypes

+1 on this one for sure.
